lib/vcap/services/api/async_requests.rb

```diff
@@ -13,38 +11,6 @@ module Api
 end
 
 module VCAP::Services::Api
-  class AsyncHttpRequest
-    class << self
-      def new(url, token, verb, timeout, msg=VCAP::Services::Api::EMPTY_REQUEST)
-        req = {
-          head: {
-            VCAP::Services::Api::GATEWAY_TOKEN_HEADER => token,
-            'Content-Type' => 'application/json'
-          },
-          body: msg.encode
-        }
-        if timeout
-          EM::HttpRequest.new(url, inactivity_timeout: timeout).send(verb.to_sym, req)
-        else
-          EM::HttpRequest.new(url).send(verb.to_sym, req)
-        end
-      end
-
-      def request(url, token, verb, timeout, msg=VCAP::Services::Api::EMPTY_REQUEST)
-        req = new(url, token, verb, timeout, msg)
-        f = Fiber.current
-        req.callback { f.resume(req) }
-        req.errback  { f.resume(req) }
-        http = Fiber.yield
-        raise UnexpectedResponse.new("Error sending request #{msg.extract.to_json} to gateway #{@url}: #{http.error}") unless http.error.empty?
-
-        code = http.response_header.status.to_i
-        body = http.response
-        [code, body]
-      end
-    end
-  end
-
   module SynchronousHttpRequest
```

Comment:
**question** I see `SynchronousHttpRequest` is used https://github.com/cloudfoundry/cloud_controller_ng/blob/main/app/models/services/managed_service_instance.rb#L99 but is `requester` method called elsewhere? Quick grep would suggest maybe not. 

```
rg requester
src/code.cloudfoundry.org/cc-uploader/handlers/upload_build_artifacts/upload_build_artifacts_test.go
131:		Context("when the requester (client) goes away", func() {

src/code.cloudfoundry.org/cc-uploader/handlers/upload_droplet/upload_droplet_test.go
178:		Context("when the requester (client) goes away", func() {

src/cloud_controller_ng/app/models/services/managed_service_instance.rb
98:    def requester

src/github.com/cloudfoundry/blobstore_url_signer/server/handler_test.go
49:			It("writes the signed URL back to requester", func() {
74:			It("writes the signed URL back to requester", func() {
```

Seems like all of the `lib/vcap/services/api` stuff is from the very old v1 service broker contract, which I dont *think* is used right?

Reply:
I didn't think about it too hard at the time because with this PR I was trying to unblock the Ruby 3.3 upgrade, but I think you're right.  I made another PR that removes the other old services code to keep the priority/purpose of this PR distinct: #4085